proto/OptimizerConfig.proto

```diff
@@ -102,22 +104,17 @@ message AdadeltaOptimizerState {
   optional TensorProto update_delta = 4;
 }
 
+
 message AdagradOptimizerState {
-  // learning rate policy
-  optional double learning_rate = 101;
-  optional double lr_decay_a = 102;
-  optional double lr_decay_b = 103;
+  optional LrPolicyState lr_state = 101;
```

Comment:
Thanks for the reminding. That's true when we maintain a project.
But what is the best practice of field number?  Making the field number auto increment by 1 seems not a smart idea. By the way, the field number is only used by `protoc` compiler to represent an entry of protobuf. Even though protobuf support indexed value by field number,  but it should not be used in user's code. Did you ever run into such a code base?